pkg/cleaner/dnats.go

```diff
@@ -0,0 +1,78 @@
+package cleaner
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/giantswarm/microerror"
+	"github.com/go-logr/logr"
+	"github.com/vmware/cloud-provider-for-cloud-director/pkg/vcdsdk"
+	capvcd "github.com/vmware/cluster-api-provider-cloud-director/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type DNATCleaner struct {
+	cli client.Client
+}
+
+func NewDNATCleaner(cli client.Client) *DNATCleaner {
+	return &DNATCleaner{cli: cli}
+}
+
+// force implementing Cleaner interface
+var _ Cleaner = &DNATCleaner{}
+
+func (lbc *DNATCleaner) Clean(ctx context.Context, log logr.Logger, vcdClient *vcdsdk.Client, c *capvcd.VCDCluster) (bool, error) {
+	log = log.WithName("DNATCleaner")
+	gateway, err := vcdsdk.NewGatewayManager(ctx, vcdClient, c.Spec.OvdcNetwork, c.Spec.LoadBalancerConfigSpec.VipSubnet)
+	if err != nil {
+		return false, err
+	}
+	if err := vcdClient.RefreshBearerToken(); err != nil {
+		return false, err
+	}
+	org, err := vcdClient.VCDClient.GetOrgByName(vcdClient.ClusterOrgName)
+	if err != nil {
+		return false, microerror.Mask(err)
+	}
+	if org == nil || org.Org == nil {
+		return false, microerror.Mask(fmt.Errorf("obtained nil org when getting org by name [%s]", vcdClient.ClusterOrgName))
+	}
+	edgeNatRules, _, err := vcdClient.APIClient.EdgeGatewayNatRulesApi.GetNatRules(
```

Comment:
@vxav can you check the logic here? I don't know the cloud-director networking in detail.

Reply:
nobody does :D I can schedule a short demo and show that it cleans what is supposed to clean if you want

Reply:
it's the number of how records you want to return. They use it the [same way](https://github.com/vmware/cloud-provider-for-cloud-director/blob/main/pkg/vcdsdk/gateway.go#L262). 

Well, if you think that more than 128 user defined services (and load balanced) will be created in the WC, I can add the paging logic here, you know a loop and using the cursor object that's part of the method signature.

Reply:
I see. Can you implement the paging with cursor as they did? I don't have an estimation of how many clusters we will have and how many services we will have per WC. I guess it will be really hard to catch this issue when we will hit that limit :D

Reply:
ok, imho overkill, but why not

Reply:
should be done in https://github.com/giantswarm/cluster-api-cleaner-cloud-director/pull/9/commits/492836c9b1e1f7c1721a59d9bf1ece1907e1bb06

price for that is the code complexity now... nested loop, breaks, but prolly right thing to do, because there is once vcd for multiple WCs... so it's not only 128 limit per 1 WC but, 128 per all WCs